src/transformers/trainer.py

```diff
@@ -1401,6 +1402,13 @@ def _inner_training_loop(
 
         model = self._wrap_model(self.model_wrapped)
 
+        if is_sagemaker_mp_enabled() and resume_from_checkpoint is not None:
+            if os.path.isfile(os.path.join(resume_from_checkpoint, WEIGHTS_NAME)):
+                self._load_from_checkpoint(resume_from_checkpoint, model)
+            else:
+                # We load the sharded checkpoint
+                load_sharded_checkpoint(model, resume_from_checkpoint)
```

Comment:
This is all done in `self._load_from_checkpoint`
```suggestion
            self._load_from_checkpoint(resume_from_checkpoint, model)
```